src/parser.rs

```diff
@@ -4227,6 +4230,33 @@ impl<'a> Parser<'a> {
         })
     }
 
+    pub fn parse_tablesample_args(&mut self) -> Result<Option<TableSample>, ParserError> {
+        if self.parse_keyword(Keyword::TABLESAMPLE) {
+            let sampling_method =
+                match self.expect_one_of_keywords(&[Keyword::BERNOULLI, Keyword::SYSTEM])? {
+                    Keyword::BERNOULLI => SamplingMethod::BERNOULLI,
+                    Keyword::SYSTEM => SamplingMethod::SYSTEM,
+                    _ => unreachable!(),
+                };
```

Comment:
You're generating an `unreachable` unnecessarily. Since the behavior differs between the variants, you can replace this by:

```suggestion
let sampling_method = if self.parse_keyword(Keyword::BERNOULLI) {
    SamplingMethod::BERNOULLI
}  else if self.parse_keyword(Keyword::SYSTEM) {
    SamplingMethod::SYSTEM
} else {
    return self.expected("BERNOULLI or SYSTEM", self.peek_token()),
}
```